components/table-of-contents/src/Section/index.js

```diff
@@ -11,32 +11,39 @@ export class Entry extends Component {
     toc: PropTypes.shape({
       addContent: PropTypes.func.isRequired,
       removeContent: PropTypes.func.isRequired,
-    }).isRequired,
+    }),
   }
 
   componentDidMount() {
-    this.props.toc.addContent(this.props.id, this.props.title);
+    if(this.props.toc) {
+      this.props.toc.addContent(this.props.id, this.props.title);
+    }
   }
 
   componentWillUnmount() {
-    this.props.toc.removeContent(this.props.id);
+    if(this.props.toc) {
+      this.props.toc.removeContent(this.props.id);
+    }
   }
 
   render() {
     return <div id={this.props.id}>{this.props.children}</div>;
   }
 }
 
-const Section = props => (
-  <Context.Consumer>
-    {toc => <Entry {...props} toc={toc} />}
-  </Context.Consumer>
-);
+const Section = ({standalone, ...props}) => {
```

Comment:
i'm not sure i see the point of `standalone`, or indeed a case for ever using it

using the flag would make the section not appear in the TOC - which could be achieved by just not using this component in the first place.

Reply:
also for destructruring there should be spaces so this should be `const Section = ({ standalone, ...props }) => {`

Reply:
This possible addition just means that if the Section is intended to be used outside the Container then the `toc` value (which would be undefined) is not sent into Entry as a prop.

Reply:
the reality of this in-use (where the bug got found) was that a component that included a section got used outside of a context.  it would require that component to be more complex were it to have to put in a `standalone` flag, as there would need to be logic to ensure that flag would only be applied in one of the places the component is used

i'm fairly certain the changes in `componentDidMount` and `componentWillUnmount` would just let this stuff work without the need for the flag anyway - so the flag feels like extra complexity that's not required

Reply:
Right, reverted that second commit. :)